### PR TITLE
SONiC Entity MIB Extension HLD

### PR DESCRIPTION
**Pull request by keboliu**
Add HLD doc for SONiC Entity MIB Extension

### CODE REVIEW

File: doc/snmp/extension-to-physical-entity-mib.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+To have this kind info available we need to extend the current platform API, can add a new function "get_contained_in_device_name" to DeviceBase class, this function will return the name of the device which it is contained in. This new field will be populated to state DB by related PMON daemons.
+
+	class DeviceBase(object):
+	    """
+	    Abstract base class for interfacing with a generic type of platform
+	    peripheral device
+	    """
+
+	    def get_contained_in_device_name(self):
```
Comment (lguohan) [inline doc/snmp/extension-to-physical-entity-mib.md:127]: can you sync with @jleveque on the need of this new platform api?

File: doc/snmp/extension-to-physical-entity-mib.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+ | 0.1 |             |      Kebo Liu      | Initial version                   |
+
+
+
+## 1. Overview 
+
+The Entity MIB contains several groups of MIB objects, currently SONiC only implemented part of the entityPhysical group following RFC2737. The group contains a single table called "entPhysicalTable" to indentify the pysical components of the system. The MIB objects of "entityPhysical" group listed as below:
+
+	EntPhysicalEntry ::= SEQUENCE {
+		entPhysicalIndex          PhysicalIndex,
```
Comment (lguohan) [inline doc/snmp/extension-to-physical-entity-mib.md:16]: can you describe the rule to calculate the index?

File: doc/snmp/extension-to-physical-entity-mib.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+	    """
+
+	    def get_contained_in_device_name(self):
+		"""
+		Retrieves The name of the device which contains this component
+		Returns:
+		    string: The name of the device
+		"""
+		raise NotImplementedError
+
```
Comment (lguohan) [inline doc/snmp/extension-to-physical-entity-mib.md:134]: please also add a section to describe the platform api you plan to use for this proposal.

File: doc/snmp/extension-to-physical-entity-mib.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+## 5. Entity MIB extension test
```
Comment (lguohan) [inline doc/snmp/extension-to-physical-entity-mib.md:135]: will a new deamon to be added to put those data in the state db?

File: doc/snmp/extension-to-physical-entity-mib.md (hunk 1 above)
Comment (jleveque) [reply doc/snmp/extension-to-physical-entity-mib.md:127]: I don't see where this `contained_in_device` string is used in the MIB. I only see `entPhysicalContainedIn`, but its type is `INTEGER`.

File: doc/snmp/extension-to-physical-entity-mib.md (hunk 2 above)
Comment (keboliu) [reply doc/snmp/extension-to-physical-entity-mib.md:17]: added in section 4.4

File: doc/snmp/extension-to-physical-entity-mib.md (hunk 4 above)
Comment (keboliu) [reply doc/snmp/extension-to-physical-entity-mib.md:281]: no new daemon needed, only need to extend current daemons to collect more data to DB which is required for the MIB

File: doc/snmp/extension-to-physical-entity-mib.md (hunk 3 above)
Comment (keboliu) [reply doc/snmp/extension-to-physical-entity-mib.md:280]: please check section 4.1

File: doc/snmp/extension-to-physical-entity-mib.md (hunk 1 above)
Comment (keboliu) [reply doc/snmp/extension-to-physical-entity-mib.md]: we can reuse the current device hierarchy defined in the platform API to deduce the contained_in relationship, removed this new API from the design.

**Comment by liat-grozovik** [comment]
this feature was already delivered as part of 2012 based on this design.

**Pull request merged**